nested-types.ipynb

```diff
@@ -0,0 +1,915 @@
+    }
+   ],
+   "source": [
+    "column_3.buffers()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.ListViewArray object at 0x108077ee0>\n",
+       "[\n",
+       "  [\n",
+       "    12,\n",
+       "    -7,\n",
+       "    25\n",
+       "  ],\n",
+       "  null,\n",
+       "  [\n",
+       "    0,\n",
+       "    -127,\n",
+       "    127,\n",
+       "    50\n",
+       "  ],\n",
+       "  [],\n",
+       "  [\n",
+       "    50,\n",
+       "    12\n",
+       "  ]\n",
+       "]"
+      ]
+     },
+     "execution_count": 28,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([   0, -127,  127,   50,   12,   -7,   25])"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "values_buffer = column_3.buffers()[4]\n",
+    "np.frombuffer(values_buffer, dtype=\"int64\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Struct\n",
+    "\n",
+    "A struct is a nested type parameterized by an ordered sequence of types.\n",
+    "\n",
+    "* one child array for each field\n",
+    "* child arrays are independent and need not be adjacent to each other in memory\n",
+    "\n",
+    "One can think of an individual struct field as a key-item pair where the key is the field and the child array its items/values. Filed (key) is saved in the schema and the items of specific field (key) are saved in the child array value buffer.\n",
+    "\n",
+    "![image info](./diagrams/struct-diagram.svg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.StructArray object at 0x1082689a0>\n",
+       "-- is_valid:\n",
+       "  [\n",
+       "    true,\n",
+       "    true,\n",
+       "    false,\n",
+       "    true,\n",
+       "    true\n",
+       "  ]\n",
+       "-- child 0 type: string\n",
+       "  [\n",
+       "    \"joe\",\n",
+       "    null,\n",
+       "    \"\",\n",
+       "    \"mark\",\n",
+       "    \"jane\"\n",
+       "  ]\n",
+       "-- child 1 type: int8\n",
+       "  [\n",
+       "    1,\n",
+       "    2,\n",
+       "    0,\n",
+       "    4,\n",
+       "    null\n",
+       "  ]"
+      ]
+     },
+     "execution_count": 31,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ty = pa.struct([pa.field('x', pa.string()),\n",
+    "                pa.field('y', pa.int8())])\n",
+    "column_4 = pa.array([(\"joe\", 1), (None, 2), None, (\"mark\", 4), (\"jane\", None)],\n",
```

Comment:
I would update this example to construct it from the list of dicts, similarly as you now updated the diagram